pytorch_lightning/trainer/connectors/logger_connector/logger_connector.py

```diff
@@ -154,6 +155,18 @@ def update_eval_step_metrics(self) -> None:
         # increment the step even if nothing was logged
         self._increment_eval_log_step()
 
+    @staticmethod
+    def _filter_metrics_for_dataloader(dl_idx, metrics, metric_prefix="dataloader_idx"):
+        result = {}
+        for k, v in metrics.items():
+            if metric_prefix not in k:
+                result[k] = v
+                continue
+            num_in_metric = int(re.search(r"\d+", k).group(0))
```

Comment:
If I have a metrics named `accuracy_class_0`, `accuracy_class_1` etc. They would also be logged only for the respective loader, right? e.g. `*_class_0` only for loader 0, `*class_1` only for loader 1 etc. 

Is that really desired behaviour?

Reply:
If you had a metrics called "accuracy_class_0" then they would be recorded for all data loaders since it is something that does not contain the prefix "dataloader_idx".

So from what I think you meant, it will not be the desired behavior like you described. Lemme know if I misunderstood :)